README.md

```diff
@@ -1,52 +1,86 @@
+The Unofficial Kik Bot API is a Python library developed to automate interactions on [Kik Messenger](https://www.kik.com).
 
-It basically lets you do the same things as the official Kik app by pretending to be a real smartphone client; It communicates with Kik's servers at `talk1110an.kik.com:5223` over a modified version of the [XMPP](https://xmpp.org/about/technology-overview.html) protocol.
+It's essentially a way to create bots that behave like humans on the platform. This library enables your bot to interact with the official Kik app by emulating a real smartphone client. It communicates with Kik's servers at `talk1110an.kik.com:5223` over a modified version of the [XMPP](https://xmpp.org/about/technology-overview.html) protocol.
+
+This library is ideal for developers, hobbyists, and businesses who want to build automated bots to interact with users, groups, and other bots on Kik.
+
+We do not endorse the use of this library for spamming or other malicious purposes. Please use this library responsibly.
 
-This is the new branch of this project and is recommended.
 ## Installation and dependencies ##
-First, make sure you are using **Python 3.6+**, not python 2.7. Second, just install it directly from GitHub:
+Make sure you have Python 3.8 or above installed on your system. You can install this library directly from GitHub:
 ```
 git clone -b new https://github.com/tomer8007/kik-bot-api-unofficial
 pip3 install ./kik-bot-api-unofficial
 ```
-## Usage ##
-Examples are a great way to understand things. A good place to start is the `examples/` directory. 
+## Quick Start Guide ##
+Here's a simple example of how to use the Kik Bot API:
 
-It is as simple as:
 ```python
 from kik_unofficial.client import KikClient
 from kik_unofficial.callbacks import KikClientCallback
 import kik_unofficial.datatypes.xmpp.chatting as chatting
 
+# Your kik login credentials (username and password)
+username = "your_kik_username"
+password = "your_kik_password"
+
+# This bot class handles all the callbacks from the kik client
 class EchoBot(KikClientCallback):
     def __init__(self):
-        self.client = KikClient(self, "your_kik_username", "your_kik_password")
+        self.client = KikClient(self, username, password)
         self.client.wait_for_messages()
 
     def on_authenticated(self):
         self.client.request_roster() # request list of chat partners
 
     def on_chat_message_received(self, chat_message: chatting.IncomingChatMessage):
-        self.client.send_chat_message(chat_message.from_jid, 'You said "{}"!'.format(chat_message.body))
+        self.client.send_chat_message(chat_message.from_jid, f'You said "{chat_message.body}"!')
+    
+if __name__ == '__main__':
+    # Creates the bot and start listening for incoming chat messages
+    callback = EchoBot()
+    client = KikClient(callback=callback, kik_username=username, kik_password=password)
```

Comment:
Why are you creating a `KikClient` both outside and inside `EchoBot`? Are you sure it's necessary?

Reply:
Not sure why, but it doesn't work if i remove it from main

Reply:
Look at `echo_bot.py` the way it is now, it works and it doesn't have this line

Reply:
Tbh I'm not sure how to fix it, I can look at it later today tho

Reply:
@tomer8007 echo_bot.py has the same issue

Reply:
I found the issue, simple_echo_bot.py stops running before the user is able to send the captcha response.. simple_echo_bot.py has to stay alive until the user submits the captcha response. so something like this would work but its kinda ugly

if __name__ == '__main__':
    # Creates the bot and start listening for incoming chat messages
    callback = EchoBot()
    time.sleep(9999999)
    
I'm sure there's a better solution but idk it

Reply:
Actually this is why I added the method `self.client.wait_for_messages()` like you can see in the current `README.md`

Reply:
yeh, but even with self.client.wait_for_messages(), it still doesn't work unless it's running in the main function. That's whats confusing me

Reply:
What's bad in running in the main function?

Reply:
I finally fixed it, wait_for_messages() makes a new thread that calls _kik_connection_thread_function(). Doing that doesn't keep the original program alive which defeats the purpose of the function. So I changed it to run _kik_connection_thread_function() without actually making a new thread and it worked

Reply:
nvm it breaks when we add logging

Reply:
@tomer8007 I fixed it again but it changed logging a bit